lib/Core/EpisodeAction/EpisodeActionReader.php

```diff
@@ -4,11 +4,61 @@
 namespace OCA\GPodderSync\Core\EpisodeAction;
 
 use InvalidArgumentException;
+use OCA\GPodderSync\Core\EpisodeAction\EpisodeActionData;
+use OCA\GPodderSync\Db\EpisodeAction\EpisodeActionRepository;
+use OCP\Http\Client\IClient;
+use OCP\Http\Client\IClientService;
+use OCP\Http\Client\IResponse;
+use OCP\ICache;
+use OCP\ICacheFactory;
 
 class EpisodeActionReader {
 	private array $requiredProperties = ['podcast', 'episode', 'action', 'timestamp'];
+    private EpisodeActionRepository $episodeActionRepository;
+    private ?ICache $cache = null;
+    private IClient $httpClient;
 
-	/**
+    public function __construct(
```

Comment:
please create a separate class for the functionality you added here and compose it with the existing class

Reply:
Might take a while until I rewrite all that. I first need to get a release of the action list on my prod server and then if I still have time I want to find out why I can't seem to be able to inject CSP settings on the admin page...

Reply:
I wasn't able to use CSP rules on admin pages, so I put together a full app page (together with lots of other stuff what would be out of scope for this a sync service) and put it in https://apps.nextcloud.com/apps/nextpod. The app depends on gpoddersync as sync service.

I'll close this PR for now.